### PR TITLE
Remove default values from records & update display annotations

### PR DESCRIPTION
**Pull request by RolandHewage**
Optional fields with default values are displayed as mandatory fields in the configuration form. This PR Remove default values from records.
Resolves https://github.com/wso2-enterprise/choreo/issues/4563

We need to improve display annotations to the new structure as in
https://docs.google.com/document/d/1-jUPEM8AzagPs-LZKY6g8mQFtqHirWcjxOA0gn6r-f4/edit?usp=sharing
Resolves https://github.com/wso2-enterprise/choreo/issues/4720

We need to fix the issue where `listEventHubs` operation doesn't have a display annotation
Resolves https://github.com/wso2-enterprise/choreo/issues/4685

Improve the display annotations following the new structure guidelines
Improve user experience in Choreo by fixing relevant issues to the connector

Remove default values from records
Improve the logic handle the above scenario
Improve the display annotations following the new structure guidelines
Add display annotations for record types
Improve documentation glitches

Remove default values from records
Add display annotations for record types

## Automation tests
 - Unit tests 
   > Done
 - Integration tests
   > Done

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

Ubuntu 20.04
JDK11
GSheet 0.99.9
azure_eventhub 0.1.4-SNAPSHOT
 
https://git-scm.com/book/en/v2/Git-Tools-Stashing-and-Cleaning
http://serebrov.github.io/html/2014-01-04-git-revert-multiple-recent-comments.html
https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions#timeoutexception
https://stackoverflow.com/questions/48173499/azure-event-hub-timeouts

### CODE REVIEW

File: eventhub/endpoint.bal [hunk 1]
```diff
@@ -82,14 +86,14 @@ public client class Client {
...
-    # + eventHubDescriptionToUpdate - event hub description to update
-    # + return - Return EventHub or Error
+    # + eventHubPath - Event Hub path (Event Hub name)
+    # + eventHubDescriptionToUpdate - EventHubDescriptionToUpdate record with properties to update
+    # + return - EventHub record on success, else returns an error
     @display {label: "Update Event Hub"}
-    remote isolated function updateEventHub(@display {label: "Event hub path"} string eventHubPath, 
-                                            @display {label: "Event hub description to update"} 
+    remote isolated function updateEventHub(@display {label: "Event Hub Path"} string eventHubPath, 
+                                            @display {label: "Event Hub Description to Update"}
```
Comment (abeykoon) [inline eventhub/endpoint.bal:96]: `Update Description` is enough i guess as the label.

File: eventhub/endpoint.bal [hunk 2]
```diff
@@ -280,24 +285,24 @@ public client class Client {
...
-    remote isolated function send(@display {label: "Event hub path"} string eventHubPath, 
-                                  @display {label: "Event data"} 
+    remote isolated function send(@display {label: "Event Hub Path"} string eventHubPath, 
+                                  @display {label: "Event Data"} 
                                   string|xml|json|byte[]|mime:Entity[]|stream<byte[], io:Error> data, 
-                                  @display {label: "User properties (Optional)"} map<string>? userProperties = (), 
-                                  @display {label: "Broker properties (Optional)"} map<anydata>? brokerProperties = (), 
-                                  @display {label: "Partition ID (Optional)"} int? partitionId = (), 
-                                  @display {label: "Publisher ID (Optional)"} string? publisherId = (), 
+                                  @display {label: "User Properties (Optional)"} map<string>? userProperties = (),
```
Comment (abeykoon) [inline eventhub/endpoint.bal:302]: No need to mention `(Optional)` on the label.

File: eventhub/types.bal [hunk 3]
```diff
@@ -16,87 +16,115 @@
...
-# + UserMetadata - user metadata
-# + PartitionCount - number of subscriptions on the Event Hub
+# + MessageRetentionInDays - Number of days to retain the events for this Event Hub
+# + Authorization - Authorization rules
+# + Status - Current status of the Event Hub
+# + UserMetadata - User metadata
+# + PartitionCount - Current number of shards on the Event Hub
+@display {label: "Event Hub Description"}
 public type EventHubDescription record {|
+    @display {label: "Message Retention In Days"}
```
Comment (abeykoon) [inline eventhub/types.bal:71]: `Message Retention (Days)`

File: eventhub/types.bal [hunk 4]
```diff
@@ -16,87 +16,115 @@
...
 # + OutgoingBytesPerSecond - Outgoing bytes per second
+@display {label: "Partition Description"}
 public type PartitionDescription record {
+    @display {label: "Size In Bytes"}
     int SizeInBytes?;
+    @display {label: "Begin Sequence Number"}
     int BeginSequenceNumber?;
+    @display {label: "End Sequence Number"}
     int EndSequenceNumber?;
+    @display {label: "Incoming Bytes Per Second"}
```
Comment (abeykoon) [inline eventhub/types.bal:125]: `Incoming Bytes (Per Sec)`

File: eventhub/types.bal [hunk 5]
```diff
@@ -16,87 +16,115 @@
...
 # + SizeInBytes - Size in bytes 
 # + BeginSequenceNumber - Begin sequence number
 # + EndSequenceNumber - End sequence number
 # + IncomingBytesPerSecond - Incoming bytes per second 
 # + OutgoingBytesPerSecond - Outgoing bytes per second
+@display {label: "Partition Description"}
 public type PartitionDescription record {
+    @display {label: "Size In Bytes"}
     int SizeInBytes?;
+    @display {label: "Begin Sequence Number"}
```
Comment (abeykoon) [inline eventhub/types.bal:121]: `Begin Seq Num`

File: eventhub/endpoint.bal (hunk 1 above)
Comment (RolandHewage) [reply eventhub/endpoint.bal]: Resolved in 4de8b34

File: eventhub/endpoint.bal (hunk 2 above)
Comment (RolandHewage) [reply eventhub/endpoint.bal]: Resolved in 4de8b34

File: eventhub/types.bal (hunk 3 above)
Comment (RolandHewage) [reply eventhub/types.bal]: Resolved in 4de8b34

File: eventhub/types.bal (hunk 4 above)
Comment (RolandHewage) [reply eventhub/types.bal]: Resolved in 4de8b34

File: eventhub/types.bal (hunk 5 above)
Comment (RolandHewage) [reply eventhub/types.bal]: Resolved in 4de8b34

**Review by abeykoon — APPROVED** [review]

**Pull request merged**